src/internal/task/etcd_service.go

```diff
@@ -63,6 +66,61 @@ func (es *etcdService) TaskCount(ctx context.Context, namespace string) (tasks i
 	return nTasks, nClaims, nil
 }
 
+func (es *etcdService) ListTaskF(ctx context.Context, namespace string, f func(string, *Task, bool) error) error {
+	etcdCols := newNamespaceEtcd(es.etcdClient, es.etcdPrefix, namespace)
+	var taskData Task
+	var claim Claim
+	etcdCols.taskCol.ReadOnly(ctx).List(&taskData, col.DefaultOptions(), func(key string) error {
+		var claimed bool
+		if taskData.State == State_RUNNING && etcdCols.claimCol.ReadOnly(ctx).Get(key, &claim) == nil {
+			claimed = true
+		}
+		return f(path.Join(namespace, key), &taskData, claimed)
+	})
+	return nil
+}
+
+func translateTaskState(state State) taskapi.State {
+	switch state {
+	case State_RUNNING:
+		return taskapi.State_RUNNING
+	case State_SUCCESS:
+		return taskapi.State_SUCCESS
+	case State_FAILURE:
+		return taskapi.State_FAILURE
+	}
+	return taskapi.State_UNKNOWN
+}
+
+func HandleList(ctx context.Context, svc Service, req *taskapi.ListTaskRequest, send func(info *taskapi.TaskInfo) error) error {
+	var marshaler jsonpb.Marshaler
+	return errors.EnsureStack(svc.ListTaskF(ctx, req.Namespace, func(key string, data *Task, claimed bool) error {
+		var state taskapi.State
+		if claimed {
```

Comment:
The way I prefer to structure stuff like this is to just do the default unconditionally, then if claimed overwrite with the claimed state. So like this:
```
state := translateTaskState(data.State)
if claimed {
  state = taskapi.State_CLAIMED
}
```

Reply:
makes sense, not sure why I wrote it the way I did